registration/models.py

```diff
@@ -376,3 +377,364 @@ def send_activation_email(self, site, request=None):
+            object.
+
+        ``site``
+            An object representing the site on which the user
+            registered; depending on whether ``django.contrib.sites``
+            is installed, this may be an instance of either
+            ``django.contrib.sites.models.Site`` (if the sites
+            application is installed) or
+            ``django.contrib.sites.requests.RequestSite`` (if
+            not). Consult the documentation for the Django sites
+            framework for details regarding these objects' interfaces.
+
+        ``request``
+            Optional Django's ``HttpRequest`` object from view.
+            If supplied will be passed to the template for better
+            flexibility via ``RequestContext``.
+        """
+
+        admin_approve_email_subject = getattr(
+            settings,
+            'ADMIN_APPROVAL_EMAIL_SUBJECT',
+            'registration/admin_approve_email_subject.txt'
+        )
+        admin_approve_email_body = getattr(
+            settings,
+            'ADMIN_APPROVAL_EMAIL_BODY',
+            'registration/admin_approve_email.txt'
+        )
+        admin_approve_email_html = getattr(
+            settings,
+            'ADMIN_APPROVAL_EMAIL_HTML',
+            'registration/admin_approve_email.html'
+        )
+
+        ctx_dict = {}
+        if request is not None:
+            ctx_dict = RequestContext(request, ctx_dict)
+
+        ctx_dict.update({
+            'user': user,
+            'user_id': user.registrationprofile.id,
+            'site': site,
+        })
+        subject = (getattr(settings, 'REGISTRATION_EMAIL_SUBJECT_PREFIX', '') +
+                   render_to_string(
+                       admin_approve_email_subject, ctx_dict))
+        # Email subject *must not* contain newlines
+        subject = ''.join(subject.splitlines())
+        from_email = getattr(settings, 'REGISTRATION_DEFAULT_FROM_EMAIL',
+                             settings.DEFAULT_FROM_EMAIL)
+        message_txt = render_to_string(admin_approve_email_body,
+                                       ctx_dict)
+
+        # try:
+        admins = getattr(settings, 'ADMINS')
+        admins = [admin[1] for admin in admins]
+        if not admins:
+            raise ImproperlyConfigured(
+                'Using the enhanced registration backend'
+                ' requires at least one admin in settings.ADMINS')
+
+        email_message = EmailMultiAlternatives(subject, message_txt,
+                                               from_email, admins)
+
+        if getattr(settings, 'REGISTRATION_EMAIL_HTML', True):
+            try:
+                message_html = render_to_string(
+                    admin_approve_email_html, ctx_dict)
+            except TemplateDoesNotExist:
+                pass
+            else:
+                email_message.attach_alternative(message_html, 'text/html')
+
+        email_message.send()
+
+    def delete_expired_users(self):
```

Comment:
Corresponding method in `RegistrationManager` has a difference. It executes `user.delete()` first and then `profile.delete()`. This failed during the tests, I was getting a `self.model.MultipleObjectsReturned( DoesNotExist: RegistrationProfile matching query does not exist)` error. I am guessing that since `RegistrationProfile` has a `ForeignKey` to the user when the user was deleted so was `RegistrationProfile` and deleting an object that extends it yielded this error. I am not 100% sure though, if you have any other ideas why this would happen I am open to hearing them.